11.4.4 version produces a breaking changes by removing `legendItemClick`.

#### Expected behaviour

Rollback breaking change produced by removing `legendItemClick`.

#### Actual behaviour

Using highcharts with typescript. The problem resides on the types.

HighCharts data: 
```
{
  chart: {
    type: 'empty',
  },
  exporting: {
    enabled: false,
  },
  title: {
    align: 'left',
    text: 'title',
    useHTML: true,
    style: {
      'font-style': 'normal',
      'font-weight': '400',
      'font-size': '16px',
      color: '#64666A',
    },
  },
  xAxis: {
    categories: [],
  },
  yAxis: {
    min: -750000,
    max: 700000,
    title: { text: '' },
  },
  plotOptions: {
    column: {
      borderColor: '#ffffff',
      borderRadius: 20,
    },
    series: {
      pointStart: 0,
      states: {
        hover: {
          enabled: false,
        },
      },
      events: {
        legendItemClick: function () {
          return false;
        },
      },
    },
    areaspline: {
      fillOpacity: 0.3,
      lineWidth: 2,
      threshold: null,
    },
  },
  tooltip: {
    formatter: function () {
      return false;
    },
  },
  credits: {
    enabled: false,
  },
  series: [
    {
      type: 'column',
      name: 'A',
      data: [],
      color: '#97C25F',
      pointWidth: 16,
    },
    {
      type: 'column',
      name: 'B',
      data: [],
      color: '#DE6F4C',
      pointWidth: 16,
    },
    {
      type: 'areaspline',
      name: 'C',
      color: '#E6E6E6',
      lineColor: '#F8D03B',
      fillColor: 'rgba(253, 239, 186, 0.5)',
      data: [],
      marker: {
        lineWidth: 0,
        lineColor: '#E6E6E6',
        fillColor: 'rgba(230, 230, 230, 1)',
      },
    },
  ],
};
```

With the attached data, the chart was working using previous patch version 11.4.3.
With the latest 11.4.4 version produces a breaking changes because `legendItemClick` has been removed, not deprecated as related on the latest changelog entry:

"Introduced new event and option [legend.events.itemClick](https://api.highcharts.com/highcharts/legend.events.itemClick), deprecated events.legendItemClick. See [#3957](https://github.com/highcharts/highcharts/issues/3957)."

#### Live demo with steps to reproduce
<!-- template: https://jsfiddle.net/highcharts/LLExL/ -->

#### Product version
Highcharts 11.4.4

#### Affected browser(s)


<!-- ASANA TASK ID (do not delete): 1207733936138345 -->

I can confirm the type is not available in 11.4.4. A Renovate PR that updates highcharts from 11.4.3 to 11.4.4 breaks in CI because of the missing type.

```
error TS2353: Object literal may only specify known properties, and 'legendItemClick' does not exist in type 'SeriesEventsOptionsObject'.
```

--- a/ts/Core/Axis/Color/ColorAxisDefaults.ts
+++ b/ts/Core/Axis/Color/ColorAxisDefaults.ts
@@ -466,7 +466,7 @@ const colorAxisDefaults: DeepPartial<ColorAxis.Options> = {
      * **Note:** This option is deprecated in favor of
      * [legend.events.itemClick](#legend.events.itemClick).
      *
-     * @deprecated
+     * @deprecated 11.4.4
      * @type       {Function}
      * @product    highcharts highstock highmaps
      * @apioption  colorAxis.events.legendItemClick

--- a/ts/Core/Legend/Legend.ts
+++ b/ts/Core/Legend/Legend.ts
@@ -1872,7 +1872,7 @@ export default Legend;
  * **Note:** This option is deprecated in favor of
  * Highcharts.LegendItemClickCallbackFunction.
  *
- * @deprecated
+ * @deprecated 11.4.4
  * @callback Highcharts.PointLegendItemClickCallbackFunction
  *
  * @param {Highcharts.Point} this
@@ -1888,7 +1888,7 @@ export default Legend;
  * **Note:** This option is deprecated in favor of
  * Highcharts.LegendItemClickEventObject.
  *
- * @deprecated
+ * @deprecated 11.4.4
  * @interface Highcharts.PointLegendItemClickEventObject
  *//**
  * Related browser event.
@@ -1927,7 +1927,7 @@ export default Legend;
  * **Note:** This option is deprecated in favor of
  * Highcharts.LegendItemClickCallbackFunction.
  *
- * @deprecated
+ * @deprecated 11.4.4
  * @callback Highcharts.SeriesLegendItemClickCallbackFunction
  *
  * @param {Highcharts.Series} this
@@ -1943,7 +1943,7 @@ export default Legend;
  * **Note:** This option is deprecated in favor of
  * Highcharts.LegendItemClickEventObject.
  *
- * @deprecated
+ * @deprecated 11.4.4
  * @interface Highcharts.SeriesLegendItemClickEventObject
  *//**
  * Related browser event.

--- a/ts/Core/Series/Series.ts
+++ b/ts/Core/Series/Series.ts
@@ -5166,7 +5166,7 @@ export default Series;
  * **Note:** This option is deprecated in favor of
  * Highcharts.LegendItemClickEventObject.
  *
- * @deprecated
+ * @deprecated 11.4.4
  * @interface Highcharts.SeriesLegendItemClickEventObject
  *//**
  * Related browser event.

--- a/ts/Core/Series/Series.ts
+++ b/ts/Core/Series/Series.ts
@@ -5150,7 +5150,7 @@ export default Series;
  * **Note:** This option is deprecated in favor of
  * Highcharts.LegendItemClickCallbackFunction.
  *
- * @deprecated
+ * @deprecated 11.4.4
  * @callback Highcharts.SeriesLegendItemClickCallbackFunction
  *
  * @param {Highcharts.Series} this

--- a/ts/Core/Series/SeriesDefaults.ts
+++ b/ts/Core/Series/SeriesDefaults.ts
@@ -969,7 +969,7 @@ const seriesDefaults: PlotOptionsOf<Series> = {
      * [legend.events.itemClick](#legend.events.itemClick).
      *
      * @type       {Highcharts.SeriesLegendItemClickCallbackFunction}
-     * @deprecated
+     * @deprecated 11.4.4
      * @context    Highcharts.Series
      * @apioption  plotOptions.series.events.legendItemClick
      */

--- a/ts/Series/Pie/PiePointOptions.d.ts
+++ b/ts/Series/Pie/PiePointOptions.d.ts
@@ -41,7 +41,7 @@ export interface PiePointEventsOptions extends PointEventsOptions {
      * **Note:** This option is deprecated in favor of
      * Highcharts.LegendItemClickEventObject.
      *
-     * @deprecated
+     * @deprecated 11.4.4
      *
      * @type {Highcharts.PointLegendItemClickCallbackFunction}
      *

--- a/ts/Series/Pie/PieSeriesDefaults.ts
+++ b/ts/Series/Pie/PieSeriesDefaults.ts
@@ -96,7 +96,7 @@ const PieSeriesDefaults: PlotOptionsOf<PieSeries> = {
      *  **Note:** This option is deprecated in favor of
      * [legend.events.itemClick](#legend.events.itemClick).
      *
-     * @deprecated
+     * @deprecated 11.4.4
      * @type       {Highcharts.PointLegendItemClickCallbackFunction}
      * @since      1.2.0
      * @product    highcharts highmaps

